src/toil/job.py

```diff
@@ -1496,6 +1531,56 @@ def run(self, fileStore):
         rValue = userFunction(*((self,) + tuple(self._args)), **self._kwargs)
         return rValue
 
+
+class PromisedRequirementFunctionWrappingJob(FunctionWrappingJob):
+    """
+    Adds child function using parent function parameters and fulfilled Promise values.
+
+    (see :class:`toil.job.FunctionWrappingJob` and :class:`toil.job.Job`)
+    """
+
+    def __init__(self, userFunction, *args, **kwargs):
+
+        self.promised_requirement = {}
+
+        # Reset any PromisedRequirement instances to default values until the promise can be fulfilled.
+        # See toil.job.PromisedRequirement
+        requirements = {'disk': '1M', 'memory': '32M', 'cores': 0.1}
+        for requirement, default in requirements.items():
+            if requirement in kwargs and isinstance(kwargs[requirement], PromisedRequirement):
```

Comment:
Replace `in` followed by use of `[]` operator with `try: [] except KeyError: pass else ...`